dbclient/HiveClient.py

```diff
@@ -161,6 +182,27 @@ def export_hive_metastore(self, ms_dir='metastore/'):
         all_dbs = self.log_all_databases(cid, ec_id, ms_dir)
         for db_name in all_dbs:
             self.log_all_tables(db_name, cid, ec_id, ms_dir, failed_metastore_log_path)
+
+        instance_profile_log_path = self._export_dir + 'instance_profiles.log'
+
+        with open(instance_profile_log_path, 'r') as iam_log:
+            for role in iam_log:
+                role_json = json.loads(role)
+                iam_role = role_json["instance_profile_arn"]
+                self.edit_cluster(cid, iam_role)
+                ec_id = self.get_execution_context(cid)
+                with open(failed_metastore_log_path, 'r') as err_log:
+                    for table in err_log:
```

Comment:
This would definitely work, but we're looping over all failed entries for every IAM role. 
N failed tables. 
M IAM roles. 
This is N*M since we're not removing the failed entries if we find an IAM role that successfully exports. 
I'd recommend parsing the error log once, load all the tables into memory (into a python set or list), then loop over the list until you run out of IAM roles. If you succeed, pop off the successful table from the list. 
Hope the list of tables isn't too large, we might need to store it in a file for improved reliability.